File: backend/app/modules/driver/schemas/finance.py
"""驾驶员财务相关 Schemas"""

from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel, Field


class DriverFinanceListItem(BaseModel):
    """费用单列表项"""

    id: int
    docNo: str
    docType: int
    isFinal: int = 0
    taskId: int
    taskNo: Optional[str] = None
    payeeName: Optional[str] = None
    plannedAmount: float
    actualAmount: Optional[float] = None
    status: int
    plannedPayTime: Optional[datetime] = None
    actualPayTime: Optional[datetime] = None
    payMethod: Optional[int] = None
    remark: Optional[str] = None


class DriverFinanceItemOut(BaseModel):
    """费用项明细"""

    id: int
    itemType: str
    itemName: Optional[str] = None
    quantity: Optional[float] = None
    unit: Optional[str] = None
    unitPrice: Optional[float] = None
    amount: float


class DriverFinanceDetail(DriverFinanceListItem):
    """费用单详情"""

    items: List[DriverFinanceItemOut] = Field(default_factory=list)
    payVoucherUrl: Optional[str] = None


class FinanceMonthlyAmount(BaseModel):
    month: str
    amount: float


class DriverFinanceSummary(BaseModel):
    """收入汇总"""

    totalIncome: float = 0
    prepaidAmount: float = 0
    supplementAmount: float = 0
    settledAmount: float = 0
    byMonth: List[FinanceMonthlyAmount] = Field(default_factory=list)


class DriverAccountOut(BaseModel):
    """驾驶员结算账户"""

    id: int
    accountType: int
    accountName: str
    accountNo: str
    balance: float = 0
    status: int = 1

File: frontend/driver-h5/src/api/finance.ts
import { get } from './request';
import type { PageResult } from './task';

export interface FinanceDocItem {
  id: number;
  docNo: string;
  docType: number;
  isFinal: number;
  taskId: number;
  taskNo?: string;
  payeeName?: string;
  plannedAmount: number;
  actualAmount?: number;
  status: number;
  plannedPayTime?: string;
  actualPayTime?: string;
  payMethod?: number;
  remark?: string;
}

export interface FinanceItem {
  id: number;
  itemType: string;
  itemName?: string;
  quantity?: number;
  unit?: string;
  unitPrice?: number;
  amount: number;
}

export interface FinanceDocDetail extends FinanceDocItem {
  items: FinanceItem[];
  payVoucherUrl?: string;
}

export interface FinanceSummary {
  totalIncome: number;
  prepaidAmount: number;
  supplementAmount: number;
  settledAmount: number;
  byMonth: { month: string; amount: number }[];
}

export interface DriverAccount {
  id: number;
  accountType: number;
  accountName: string;
  accountNo: string;
  balance: number;
  status: number;
}

export interface FinanceListQuery {
  page?: number;
  pageSize?: number;
  docType?: number;
  status?: number;
  yearMonth?: string;
}

export function listMyFinance(params: FinanceListQuery) {
  return get<PageResult<FinanceDocItem>>('/finance/my', params as Record<string, unknown>);
}

export function getFinanceDetail(docId: number) {
  return get<FinanceDocDetail>(`/finance/${docId}`);
}

export function getFinanceSummary(params?: { yearMonth?: string }) {
  return get<FinanceSummary>('/finance/summary', params as Record<string, unknown>);
}

export function listMyAccounts() {
  return get<DriverAccount[]>('/finance/account');
}

/** 资金账户（往来账） */
export interface FundAccount {
  id: number;
  driverId: number;
  balance: number;
  frozenAmount: number;
  totalIn: number;
  totalOut: number;
  status: number;
  lastTxnAt?: string;
}

export interface FundTransaction {
  id: number;
  txnNo: string;
  bizType: number;
  direction: number;
  amount: number;
  delta: number;
  balanceAfter: number;
  operatorName?: string;
  remark?: string;
  createdAt: string;
}

export function getMyFundAccount() {
  return get<FundAccount>('/finance/fund-account');
}

export function listMyFundTransactions(params: {
  page?: number;
  pageSize?: number;
  bizType?: number;
}) {
  return get<PageResult<FundTransaction>>(
    '/finance/fund-account/transactions',
    params as Record<string, unknown>
  );
}
